src/components/site-header/SiteHeaderLogos.jsx

```diff
@@ -1,14 +1,14 @@
 import React, { useContext } from 'react';
-import { Link, useLocation } from 'react-router-dom';
+import { Link, useRouteMatch } from 'react-router-dom';
 import edXLogo from '@edx/brand/logo.svg';
 import { AppContext } from '@edx/frontend-platform/react';
-
-import GetSmarterLogo from '../../assets/icons/getsmarter-logo.svg';
+import { COURSE_TYPE_PARTNER_LOGOS } from '../course/data/constants';
 
 const SiteHeaderLogos = () => {
+  const courseTypeMatch = useRouteMatch('/:enterpriseSlug/:courseType');
+  const courseType = courseTypeMatch?.params?.courseType;
   const { enterpriseConfig } = useContext(AppContext);
-  const { pathname } = useLocation();
-  const isInExecutiveEducationPath = pathname.includes('executive-education-2u');
+  const courseTypePartnerLogo = courseType && COURSE_TYPE_PARTNER_LOGOS[courseType];
 
   let mainLogo = (
```

Comment:
I believe it might be worth simplifying the existing styles/implementation by utilizing `Stack` from Paragon ([docs](https://paragon-openedx.netlify.app/components/stack/#horizontal-direction)).

**Current**
* The spacing between the left logo and the vertical divider compared to the spacing between the vertical divider and the right logo is not consistent.
* There is no spacing between the GetSmarter logo and the nav items, which puts the hover state styles of the first nav item right up against the GetSmarter logo which is a bit tight.

<img width="477" alt="image" src="https://user-images.githubusercontent.com/2828721/235651120-4b169218-31e3-4411-a715-268bdf738a76.png">

**Proposed**

* Uses `Stack`.
* Removes unnecessary `div`
* Treats the vertical divider as a standalone element rather than a border on a logo.
* Removes custom spacing via `margin(-*)` and/or the spacing CSS utility classes.
* Scales the GetSmarter logo by height rather than width to ensure heights stay consistent (via `$header-logo-height-desktop` and `$header-logo-height-mobile` SCSS variables).

<img width="472" alt="image" src="https://user-images.githubusercontent.com/2828721/235660555-12335b37-9874-420f-919d-829c75c7656c.png">

<img width="755" alt="image" src="https://user-images.githubusercontent.com/2828721/235660644-b856ea67-41ca-4091-8350-9227746b2ed5.png">

See this diff for the proposed changes: https://github.com/openedx/frontend-app-learner-portal-enterprise/pull/725